# FOOD BALANCE SHEETS E AFRICA NOFLAG 

## Import libraries 

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import warnings

## Read dataset

In [3]:
food = pd.read_csv(r"C:\Users\Akunna Anyamkpa\Documents\HAMOYE\PROJECT 1\FoodBalanceSheets_E_Africa_NOFLAG.csv")

## Check number of columns and rows

In [4]:
food.shape

(60943, 12)

In [5]:
food.head(4)

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   

                                  Element             Unit    Y2014     Y2015  \
0           Total Population - Both sexes     1000 persons  38924.0  39728.00   
1                Domestic supply quantity      1000 tonnes      0.0      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.0   3379.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.9     94.35   

      Y2016     Y2017     Y2018  
0  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00  
2   3372.00   3341.00   3322.00  
3     94.72     92.82     91.83

In [6]:
food.columns

Index(['Area Code', 'Area', 'Item Code', 'Item', 'Element Code', 'Element',
       'Unit', 'Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018'],
      dtype='object')

## Statistical description

In [7]:
food.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [8]:
food.dtypes

Area Code         int64
Area             object
Item Code         int64
Item             object
Element Code      int64
Element          object
Unit             object
Y2014           float64
Y2015           float64
Y2016           float64
Y2017           float64
Y2018           float64
dtype: object

# What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

In [9]:
food_an_fat = food.loc[food["Item"] == "Animal fats"] 
display(food_an_fat.tail(4))

Area Code      Area  Item Code         Item  Element Code  \
60698        181  Zimbabwe       2946  Animal fats           645   
60699        181  Zimbabwe       2946  Animal fats           664   
60700        181  Zimbabwe       2946  Animal fats           674   
60701        181  Zimbabwe       2946  Animal fats           684   

                                      Element             Unit  Y2014  Y2015  \
60698     Food supply quantity (kg/capita/yr)               kg   0.69   2.88   
60699           Food supply (kcal/capita/day)  kcal/capita/day  21.00  18.00   
60700  Protein supply quantity (g/capita/day)     g/capita/day   0.04   0.04   
60701      Fat supply quantity (g/capita/day)     g/capita/day   2.33   2.03   

       Y2016  Y2017  Y2018  
60698   6.51   2.51   2.49  
60699  18.00  20.00  19.00  
60700   0.04   0.03   0.03  
60701   2.01   2.18   2.10

In [10]:
food_an_fat_14 = food_an_fat["Y2014"].sum()
food_an_fat_14

209460.54

In [11]:
food_an_fat_17 = food_an_fat["Y2017"].sum()
food_an_fat_17

269617.53

### Answer: 209460.54, 269617.53

# What is the mean and stanard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [12]:
Mean_Y2015 = food["Y2015"].mean()
round(Mean_Y2015, 3)

135.236

In [13]:
Sd_Y2015 = food["Y2015"].std()
round(Sd_Y2015, 3)

1603.404

### Answer: 135.236, 1603.404

# What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [14]:
Total_count_2016 = len(food["Y2016"])
Total_count_2016

60943

In [15]:
Null_data_2016 = food["Y2016"].isnull().sum()
Null_data_2016

1535

In [16]:
perc_2016 = ( Null_data_2016 / Total_count_2016 ) * 100
round(perc_2016, 2)

2.52

### Answer: 1535, 2.52

# Which year had the highest correlation with 'Element Code'?

In [17]:
corr = food.corr()
corr

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

### Answer: 2014

# What year has the highest sum of import Quantity from 2014 up till 2017

In [18]:
food_im_qty = food.loc[food["Element"] == "Import Quantity"] 
food_im_qty.head(4)

Area Code     Area  Item Code                      Item  Element Code  \
12          4  Algeria       2905  Cereals - Excluding Beer          5611   
28          4  Algeria       2511        Wheat and products          5611   
44          4  Algeria       2805         Rice and products          5611   
59          4  Algeria       2513       Barley and products          5611   

            Element         Unit    Y2014    Y2015    Y2016    Y2017    Y2018  
12  Import Quantity  1000 tonnes  12604.0  13983.0  13523.0  13039.0  13182.0  
28  Import Quantity  1000 tonnes   7461.0   8531.0   8255.0   8105.0   8434.0  
44  Import Quantity  1000 tonnes    171.0    164.0    162.0    190.0    207.0  
59  Import Quantity  1000 tonnes    805.0    797.0    916.0    567.0    402.0

In [19]:
sum_14 = food.Y2014.sum() 
sum_14

7965086.130000001

In [20]:
sum_15 = food.Y2015.sum() 
sum_15

8032340.2299999995

In [21]:
sum_16 = food.Y2016.sum() 
sum_16

8112472.649999999

In [22]:
sum_17 = food.Y2017.sum() 
sum_17

8375729.189999999

### Answer: 2017

# What is the total number of the sum of Production in 2014?

In [23]:
food_prod = food.loc[food["Element"] == "Production"] 
food_prod.head(5)

Area Code     Area  Item Code                      Item  Element Code  \
11          4  Algeria       2905  Cereals - Excluding Beer          5511   
27          4  Algeria       2511        Wheat and products          5511   
43          4  Algeria       2805         Rice and products          5511   
58          4  Algeria       2513       Barley and products          5511   
73          4  Algeria       2514        Maize and products          5511   

       Element         Unit   Y2014   Y2015   Y2016   Y2017   Y2018  
11  Production  1000 tonnes  3436.0  3761.0  3445.0  3478.0  6065.0  
27  Production  1000 tonnes  2436.0  2657.0  2440.0  2437.0  3981.0  
43  Production  1000 tonnes     0.0     0.0     0.0     0.0     0.0  
58  Production  1000 tonnes   939.0  1031.0   920.0   970.0  1957.0  
73  Production  1000 tonnes     3.0     3.0     4.0     3.0     4.0

In [24]:
food_prod.Y2014.sum()

1931287.75

### Answer: 1931287.75

# Which of these elements had the highest sum in 2018?

In [25]:
food_fsq = food.loc[food["Element"] == "Food supply quantity (kg/capita/yr)"] 
food_fsq.Y2018.sum()

49056.85

In [26]:
food_iq = food.loc[food["Element"] == "Import Quantity"] 
food_iq.Y2018.sum()

287997.08999999997

In [27]:
food_dsq = food.loc[food["Element"] == "Domestic supply quantity"] 
food_dsq.Y2018.sum()

2161192.1

In [28]:
food_fs = food.loc[food["Element"] == "Food supply (kcal/capita/day)"] 
food_fs.Y2018.sum()

455261.0

### Answer: Food supply quantity (kg/capita/yr)

# Which of these elements had the 3rd lowest sum in 2018?

In [29]:
food_seed = food.loc[food["Element"] == "Seed"] 
food_seed.Y2018.sum()

25263.14

In [30]:
food_prot = food.loc[food["Element"] == "Protein supply quantity (g/capita/day)"] 
food_prot.Y2018.sum()

11833.56

In [31]:
food_fat = food.loc[food["Element"] == "Fat supply quantity (g/capita/day)"] 
food_fat.Y2018.sum()

10258.689999999999

### Answer: Seed

# What is the total Import Quantity in Algeria in 2018?

In [32]:
food_im_qty_alg = food_im_qty.loc[food_im_qty["Area"] == "Algeria"] 
food_im_qty_alg.Y2018.sum()

36238.29

### Answer: 36238.29

# What is the total number of unique countries in the dataset?

In [33]:
countries_un = food.Area.unique()

In [34]:
len(countries_un)

49

### Answer: 49